rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1695,9 +1683,6 @@ static bool dds_qos_to_rmw_qos(const dds_qos_t * dds_qos, rmw_qos_profile_t * qo
       case DDS_LIVELINESS_AUTOMATIC:
         qos_policies->liveliness = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC;
         break;
-      case DDS_LIVELINESS_MANUAL_BY_PARTICIPANT:
```

Comment:
Removing this case outright is a problem: this conversion routine is also used by the functions that monitor remote publications/subscriptions to feed it into the graph cache, and so `DDS_LIVELINESS_MANUAL_BY_PARTICIPANT` may still show up in its input.

Ignoring the endpoint altogether would be possible but seems unwise (it might be intentional in a more complex setting where there are some non-ROS2 processes involved), so it needs to be mapped in some sensible way. Given the desire to remove `BY_NODE` and the (in my view) undesirability of adding the cop-add of an `OTHER` mode, mapping it to `AUTOMATIC` seems reasonable.

Reply:
does using `RMW_QOS_POLICY_LIVELINESS_UNKNOWN` sound reasonable?

Reply:
Yes, that sounds perfectly reasonable. I hadn't realized it already existed 🙂